### PR TITLE
Fix some integer over/under flow

### PR DESCRIPTION
**Pull request by holygits**
Fix missing ? operator on fallible `input.read`

### CODE REVIEW

**Comment by HaoZhaiX** [comment]
LGTM

File: src/lib.rs [hunk 1]
```diff
@@ -131,7 +129,7 @@ impl Encode for Method {
 
         if let Some(constraints) = &self.constraints {
             #[allow(clippy::cast_possible_truncation)]
```
Comment (elim0322) [inline src/lib.rs]: Do we still need this attribute? looks a bit weird inline

File: src/lib.rs [hunk 2]
```diff
@@ -223,7 +221,7 @@ impl Encode for CENNZnutV0 {
         buf.write(&[0, 0]);
 
         #[allow(clippy::cast_possible_truncation)]
```
Comment (elim0322) [inline src/lib.rs]: could we remove this attribute if possible

File: src/lib.rs (hunk 1 above)
Comment (holygits) [reply src/lib.rs]: Prefer leave it for now, its for the lossy `as` cast

**Pull request merged**